tests/units/hindawi/test_hindawi_parser.py

```diff
@@ -187,3 +244,60 @@ def test_hindawi_parsing(parsed_articles, expected, key):
         article,
     ) in zip(expected, parsed_articles):
         assert article[key] == expected_value
+
+
+@pytest.fixture()
+def articles_with_missing_data(shared_datadir):
+    articles = {}
+    files = ["without_doi.xml", "without_page_nr.xml"]
+
+    for file in files:
+        with open(shared_datadir / file) as f:
+            articles[file] = ET.fromstring(f.read())
+    return articles
+
+
+def test_parse_articles(hindawi_parser, articles_with_missing_data):
+    for file_name_with_missing_data in articles_with_missing_data.keys():
+        # DOI is required, it has to crash in the beginning
+        if file_name_with_missing_data == "without_doi.xml":
+            with pytest.raises(RequiredFieldNotFoundExtractionError):
+                hindawi_parser._publisher_specific_parsing(
+                    articles_with_missing_data[file_name_with_missing_data]
+                )
+        else:
+            hindawi_parser._publisher_specific_parsing(
+                articles_with_missing_data[file_name_with_missing_data]
+            )
```

Comment:
it's better not to add additional logic to the tests, it would be better to create two separate tests. Also the `else` doesn't test anything :)

Reply:
it does because this test function doesn't have the return sentence. So the crashing one still we be executed on the line 269

Reply:
Then we only need to test the case that raises `RequiredFieldNotFoundExtractionError`

Reply:
yes, I agree